### DISCUSSION TITLE
Custom penalties for PELT

### DISCUSSION THREAD

**Discussion by rnagabhyrava** [deepcharles/ruptures#166] [Q&A] [answered]
Hey guys,

First of all, I am new to changepoint detection and this package has helped me a lot. Love your work and thank you for all the time and effort you put into this.

Coming to my question, I am working with PELT and various costs (l1, l2, normal). I am confused about what penalties to use with for each of these. I believe we can use AIC/BIC (correct me if wrong), [as you mentioned here](https://github.com/deepcharles/ruptures/issues/4) this bic formula only applies for l2. Can someone help me  with what penalty/formula goes with each of l1, l2 and normal with PELT?

Thanks,
rnaga

**Comment by deepcharles** [comment, marked as answer]
Sorry for the late reply.

To compute the BIC criterion, you need to plug in the [definition](https://en.wikipedia.org/wiki/Bayesian_information_criterion) the correct likelihood function and maximum likelihood estimator.

Using the Gaussian likelihood (with fixed variance) yields the BIC formula for the `L2` cost function.
Using the Gaussian likelihood (with varying variance) yields the BIC formula for the `normal` cost function.
You can refer to [this book](https://link.springer.com/book/10.1007/978-0-8176-4801-5) for the computation of the maximum likelihood for several likelihood functions.

In BIC, the number of parameters (denoted `k` in the Wikipedia formula) is usually proportional to the number of change points and the signal dimension.

As for the L1 cost function, I am not sure there is a BIC that can be derived.

Hope that helps
